src/main/java/org/bahmni/reports/template/TSIntegrationDiagnosisReportTemplate.java

```diff
@@ -0,0 +1,145 @@
+
+        return new BahmniReportBuilder(jasperReport);
+    }
+
+    private void loadTempTable(Connection connection, String tempTableName, String parentCode) {
+        int offset = 0;
+        int pageSize = getDefaultPageSize();
+        try {
+            String createSqlStmt = MessageFormat.format(CREATE_SQL_TEMPLATE, tempTableName);
+            String insertSqlStmt = MessageFormat.format(INSERT_SQL_TEMPLATE, tempTableName);
+            Statement statement = connection.createStatement();
+            statement.execute(createSqlStmt);
+            PreparedStatement pstmtInsert = connection.prepareStatement(insertSqlStmt);
+
+            TSPageObject pageObject = null;
+            do {
+                try {
+                    pageObject = fetchDescendantsByPagination(parentCode, pageSize, offset, "en");
+                } catch (IOException e) {
+                    throw new RuntimeException();
+                }
+                List<String> codes = pageObject.getCodes();
+                for (int batchcount = 0; batchcount < codes.size(); batchcount++) {
+                    pstmtInsert.setString(1, codes.get(batchcount));
+                    pstmtInsert.addBatch();
+                }
+                pstmtInsert.executeBatch();
+                offset += pageSize;
+            } while (offset < pageObject.getTotal());
+        } catch (SQLException e) {
+            logger.error("Error occured while making database call to " + tempTableName + " table");
+            throw new RuntimeException();
+        }
+    }
+
+    private TSPageObject fetchDescendantsByPagination(String terminologyCode, int pageSize, int offset, String localeLanguage) throws IOException {
+        String url = MessageFormat.format(descendantsUrlTemplate, terminologyCode, pageSize, offset, localeLanguage);
+        String responseStr = httpClient.get(URI.create(url));
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            return mapper.readValue(responseStr, TSPageObject.class);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException();
+        }
+    }
+
+
+    private String getFormattedSql(String templateSql, String conceptSourceCode, String startDate, String endDate, String tempTableName) {
+        ST sqlTemplate = new ST(templateSql, '#', '#');
+        sqlTemplate.add("conceptSourceCode", conceptSourceCode);
+        sqlTemplate.add("startDate", startDate);
+        sqlTemplate.add("endDate", endDate);
+        sqlTemplate.add("tempTable", tempTableName);
+        return sqlTemplate.render();
+    }
+
+
+    public int getDefaultPageSize() {
+        String pageSize = System.getenv("REPORTS_TS_PAGE_SIZE");
+        if (pageSize == null)
+            pageSize = tsProperties.getProperty("ts.defaultPageSize");
+        if (pageSize != null) {
+            return Integer.parseInt(pageSize);
+        }
+        return 20;
```

Comment:
This is a Magic number. Should be a constant on TOP of the class file.

Reply:
We have made the default page size as constant